pkg/tfgen/examples_coverage_exporter.go

```diff
@@ -0,0 +1,158 @@
+// and a reference to the CoverageTracker that created it
+type CoverageExportUtil struct {
+	CT *CoverageTracker // Reference to the Coverage Tracker that wants to turn its data into a file
+}
+
+func newCoverageExportUtil(coverageTracker *CoverageTracker) CoverageExportUtil {
+	return CoverageExportUtil{coverageTracker}
+}
+
+// The entire export utility interface. Will attempt to export the Coverage Tracker's data into the
+// specified output directory, and will panic if an error is encountered along the way
+func (CE *CoverageExportUtil) tryExport(outputDirectory string) {
+	CE.exportUploadableResults(outputDirectory, "summary.json")
+	CE.exportSummarizedResults(outputDirectory, "concise.json")
+}
+
+// Three different ways to export coverage data:
+// The first mode, using a large provider > example map
+func (CE *CoverageExportUtil) exportFullResults(outputDirectory string, fileName string) {
+
+	// The Coverage Tracker data structure remains identical, the only thing added in the file is the name of the provider
+	ProviderNameToExamplesMap := map[string]map[string]GeneralExampleInfo{CE.CT.ProviderName: CE.CT.EncounteredExamples}
+
+	jsonOutputLocation := createJsonOutputLocation(outputDirectory, fileName)
+	marshalAndWriteJson(ProviderNameToExamplesMap, jsonOutputLocation)
+}
+
+// The second mode, similar to existing Pulumi coverage Json files uploadable to redshift
+func (CE *CoverageExportUtil) exportUploadableResults(outputDirectory string, fileName string) {
+
+	// The Coverage Tracker data structure is flattened down to the example level, and they all
+	// get individually written to the file in order to not have the "{ }" brackets at the start and end
+	type SingleExampleResult struct {
+		ProviderName    string
+		ProviderVersion string
+		ExampleName     string
+		FailedLanguages []LanguageConversionResult
+	}
+
+	jsonOutputLocation := createJsonOutputLocation(outputDirectory, fileName)
+
+	// All the examples in the map are iterated by key and marshalled into one large byte array
+	// separated by \n, making the end result look like a bunch of Json files that got concatenated
+	var result = []byte{}
+	for _, exampleInMap := range CE.CT.EncounteredExamples {
+		singleExample := SingleExampleResult{
+			ProviderName:    CE.CT.ProviderName,
+			ProviderVersion: CE.CT.ProviderVersion,
+			ExampleName:     exampleInMap.Name,
+			FailedLanguages: []LanguageConversionResult{},
+		}
+
+		for _, conversionResult := range exampleInMap.LanguagesConvertedTo {
+			if conversionResult.FailureSeverity > 0 {
+				singleExample.FailedLanguages = append(singleExample.FailedLanguages, conversionResult)
+			}
+		}
+		marshalledExample, err := json.MarshalIndent(singleExample, "", "\t")
+		panicIfError(err, "Failed to MarshalIndent JSON file")
+		result = append(append(result, marshalledExample...), uint8('\n'))
+	}
+	err2 := ioutil.WriteFile(jsonOutputLocation, result, 0600)
```

Comment:
You can just reuse the existing error:

```suggestion
	err = ioutil.WriteFile(jsonOutputLocation, result, 0600)
```